packages/api-client/src/exportData.js

```diff
@@ -0,0 +1,218 @@
+
+    if (fs.lstatSync(filePath).isDirectory()) {
+      rmdirRecursive(filePath);
+    } else {
+      fs.unlinkSync(filePath);
+    }
+  });
+
+  fs.rmdirSync(pathToDir);
+}
+
+/**
+ * Saves locally non-public files in order to be able to import them in the future
+ */
+function workWithFiles(files: Array<*>, pathToSave: string) {
+  const promises = [];
+  let isFolderCreated = false;
+
+  for (const file of files) {
+    changeKeysName({ downloadUrl: 'url' }, 1, file);
+
+    if (!file.public && file.filename && file.filename.length > 0) {
+      // if we found first non public file then remove old folder and create a new one
+      if (!isFolderCreated) {
+        if (fs.existsSync(pathToSave)) {
+          rmdirRecursive(pathToSave);
+        }
+
+        // $FlowFixMe options were added in nodejs 10.x so we need to update flow
+        fs.mkdirSync(pathToSave, { recursive: true });
+        isFolderCreated = true;
+      }
+
+      const filePath = path.join(pathToSave, file.filename);
+      const fileStream = fs.createWriteStream(filePath);
+
+      promises.push(new Promise((resolve, reject) => {
+        https.get(file.url, response => {
+          response.pipe(fileStream);
+          file.url = path.resolve(filePath);
+          resolve();
+        }).on('error', () => {
+          fs.unlinkSync(filePath);
+          reject();
+        });
+      }));
+    }
+  }
+
+  return Promise.all(promises);
+}
+
+export const exportData = async (
+  request: (query: string | DocumentNode, variables?: Object) => Promise<Object>,
+  config: ExportDataConfig = DEFAULT_CONFIG,
+) => {
+  let schemaTables = await exportTables(request, config);
+
+  // Fetch Files table schema if we export data without system tables
+  if (!config.withSystemTables) {
+    const filesSchemaTable = await request(TABLES_LIST_QUERY, {
+      filter: { tableNames: [SYSTEM_TABLES.FILES] },
+    });
+
+    if (R.hasPath(['tablesList', 'items'], filesSchemaTable)) {
+      schemaTables = [...schemaTables, ...filesSchemaTable.tablesList.items];
+    }
+  }
+
+  const result = {
+    tables: schemaTables,
+    data: {},
+  };
+
+  for (const table of schemaTables) {
+    const listFields = [];
+    const { name } = table;
+    let { fields } = table;
+    let queryBody = '';
+
+    if (isFilesTable(table)) {
+      fields = fields.filter(item => FILE_FIELDS.includes(item.name));
+    }
+
+    for (const field of fields) {
+      if (!field.relation && !isFileField(field)) {
+        queryBody = `${queryBody}${field.name}\n`;
+      } else {
+        queryBody = field.isList
+          ? listFields.push(field.name) &&
+            `${queryBody}${field.name} {
+                items {
+                  id
+                }
+              }\n`
+          : `${queryBody}${field.name} {
+                id
+              }\n`;
+      }
+    }
+
+    const queryFieldName = `${lowerFirstLetter(name)}List`;
+    const tableItems = await request(`
```

Comment:
@lesha1201  for generating queries u should use special functions from `@8base/schema-name-generator` or u can use query generators from the `@8base/utils`, because current solution it doesn't work for the table with `untitledTable` name for example

Reply:
@zouxuoz Fixed